src/middlewared/middlewared/plugins/mdns.py

```diff
@@ -89,49 +120,201 @@ def start_mdnsd(self):
         return p.returncode == 0
 
 
-class mDNSThread(threading.Thread):
+class mDNSServiceObject(object):
     def __init__(self, **kwargs):
-        super(mDNSThread, self).__init__()
-        self.setDaemon(True)
-        self.middleware = kwargs.get('middleware')
-        self.logger = self.middleware.logger
-        self.timeout = kwargs.get('timeout', 30)
+        self.sdRef = kwargs.get('sdRef')
+        self.txtrecord = kwargs.get('txtrecord')
+        self.port = kwargs.get('port')
+        self.regtype = kwargs.get('regtype')
+        self.name = kwargs.get('name')
 
-    def active(self, sdRef):
-        return (bool(sdRef) and sdRef.fileno() != -1)
+    @property
+    def fullname(self):
+        return "%s.%s.%s" % (
```

Comment:
There are three %s and only two args.
Also lets use .format() or f-string please

Reply:
I'll do better and replace where I was using this with a simple dictionary.